prototypes/marimekko-chart/src/main.js

```diff
@@ -0,0 +1,685 @@
+            labelMode: labelMode,
+            labelBars: labelBars,
+            labelSeg: labelSeg,
+            numeric: numeric,
+            percentage: percentage,
+            category: category,
+            reverseBars: reverseBars,
+            reverseSeg: reverseSeg,
+            sortBar: sortBar,
+            sortSeg: sortSeg
+        };
+        
+        /**
+         * Check for any errors.
+         */
+        let errors = await dataView.getErrors();
+        if (errors.length > 0) {
+            mod.controls.errorOverlay.show(errors, "dataView");
+            // TODO clear DOM
+            return;
+        }
+
+        mod.controls.errorOverlay.hide("dataView");
+
+        // Return and wait for next call to render when reading data was aborted.
+        // Last rendered data view is still valid from a users perspective since
+        // a document modification was made during a progress indication.
+        // Hard abort if row count exceeds an arbitrary selected limit
+        const xLimit = 1000;
+        const colorLimit = 100;
+        const colorCount = (await dataView.hierarchy("Color")).leafCount;
+        const xCount = (await dataView.hierarchy("Categories")).leafCount;
+        if (colorCount > colorLimit || xCount > xLimit) {
+            canvasDiv.remove();
+                        mod.controls.errorOverlay.show("The resulting data view exceeded the size limit.", "dataViewSize1");
+            if (xCount > xLimit) {
+                mod.controls.errorOverlay.show(
+                    `Maximum number of categories is limited to ${xLimit}. Try aggregating the expression further.`,
+                    "dataViewSize2"
+                );
+            } else {
+                mod.controls.errorOverlay.hide("dataViewSize2");
+            }
+                        if (colorCount > colorLimit) {
+                mod.controls.errorOverlay.show(
+                    `Maximum number of colors is limited to ${colorLimit} for readability reasons.`,
+                    "dataViewSize3"
+                );
+            } else {
+                mod.controls.errorOverlay.hide("dataViewSize3");
+            }
+                        return;
+        } else {
+            mod.controls.errorOverlay.hide("dataViewSize1");
+            mod.controls.errorOverlay.hide("dataViewSize2");
+            mod.controls.errorOverlay.hide("dataViewSize3");
+        }
+
+        //Clear marking of segments
+        modContainer.onclick = (e) => {
+            if (e.target === xTitleDiv ||
+                e.target === modContainer ||
+                e.target === canvasDiv ||
+                e.target === scale) {
+                dataView.clearMarking();
+            }
+        };
+
+        // Get the leaf nodes for the category hierarchy. We will iterate over them to
+        // render the bars.
+        let Hierarchy = await dataView.hierarchy("Categories");
+        let Root = await Hierarchy.root();
+        if (Root == null) {
```

Comment:
A personal preference, but these names do not specify which hiearchy that the variable represents. The color axis is also a hierarchy in this mod.

Also, the mod mixes pascal and camel casing.

Reply:
should be fixed now but if you find any more pascal cases let me know!